lib/rust/parser/src/macros/resolver.rs

```diff
@@ -186,43 +195,61 @@ enum Step<'s> {
     MacroStackPop(syntax::Item<'s>),
 }
 
+/// Information about macro resolution state that is stored while processing a deeper indentation
+/// level.
+#[derive(Debug)]
+struct Scope<'s> {
+    parent_tokens: std::vec::IntoIter<syntax::Item<'s>>,
+    macros_start:  usize,
+    outputs_start: usize,
```

Comment:
I dont understand what these numbers are. I've read the docs, but its still not clear to me. I mean, I understand it, but its hard to understand it without our domain knowledge. Could you please add docs to Scope or to Resolver (even better) explaining how nested macros are resolved in case of blocks, please? It's worth explaining that blocks have precedence over macros. For example, 

```
if foo
    then bar
```

would not parse as "if ... then ..." - it will parse as "if foo" (not fully matched macro) applied to a code block. It's important to describe that in the docs - either here or in the global docs in the parser module.